index.js

```diff
@@ -14,6 +14,7 @@ module.exports.pitch = function(remainingRequest) {
 		'		if(!component) {',
 		'			require.ensure([], function() {',
 		'				component = require(' + loaderUtils.stringifyRequest(this, moduleRequest) + ');',
+		'				if (typeof component === "object") component = component.default;',
```

Comment:
lets update this to just `if (component && component.default) component = component.default;`

This will let component be any value, but if it is non-null and has a 'default' key it will take it over the original.

This being targeted towards React Components the 'component' value does not need to be as flexible as other loaders. We could even add a warning if the component isn't "a react renderable component"